File: itx-go/itx-contract/queue/factory/factory.go
package factory

import (
	"github.com/chehsunliu/itx/itx-go/itx-contract/queue"
)

type MessageQueueFactory interface {
	CreateControlStandardQueue() queue.MessageQueue
	CreateControlPremiumQueue() queue.MessageQueue
	CreateComputeStandardQueue() queue.MessageQueue
	CreateComputePremiumQueue() queue.MessageQueue
}

File: itx-rs/itx-contract/src/queue/factory.rs
use std::sync::Arc;

use crate::queue::MessageQueue;

pub trait MessageQueueFactory: Send + Sync {
    fn create_control_standard_queue(&self) -> Arc<dyn MessageQueue>;
    fn create_control_premium_queue(&self) -> Arc<dyn MessageQueue>;
    fn create_compute_standard_queue(&self) -> Arc<dyn MessageQueue>;
    fn create_compute_premium_queue(&self) -> Arc<dyn MessageQueue>;
}
